src/silx/utils/array_like.py

```diff
@@ -49,6 +49,9 @@
 import numpy
 import numbers
 
+from .._utils import NP_OPTIONAL_COPY
```

Comment:
Do you use it as a flag? In case of troubles, you can switch it off?

Not sure it's a very good idea. Because it is spread everywhere. Isn't it better to put it in this module, only?

Reply:
`NP_OPTIONAL_COPY` was added to support both numpy v1 and v2 (PR #4082).

It was done following [numpy2 migration guide](https://numpy.org/doc/stable/numpy_2_0_migration_guide.html#adapting-to-changes-in-the-copy-keyword).

The day `silx` stop supporting numpy v1 it will also help a lot to clean-up the code.